server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -2258,6 +2264,20 @@ public interface Warmer {
      */
     public abstract void deactivateThrottling();
 
+    /**
+     * If indexing is throttled to the point where it is paused completely,
+     * another task trying to get indexing permits might want to pause throttling
+     * by letting one thread pass at a time so that it does not get starved.
+     */
+    public abstract void suspendThrottling();
+
+    /**
+     * Reverses a previous {@link #suspendThrottling} call.
+     */
+    public abstract void resumeThrottling();
+
+    public abstract boolean isIndexingPaused();
```

Comment:
I believe my comment above could remove the last usage of this - in that case we should remove the method too.